src/dyncfg-ld/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A dyncfg::ConfigSet backed by LaunchDarkly.
+
+use std::time::Duration;
+
+use launchdarkly_server_sdk as ld;
+use mz_build_info::BuildInfo;
+use mz_dyncfg::{ConfigSet, ConfigUpdates, ConfigVal};
+use mz_ore::cast::CastLossy;
+use mz_ore::future::TimeoutError;
+use mz_ore::task;
+use tokio::time;
+
+/// Start a background task that syncs to a ConfigSet from LaunchDarkly. `ctx_builder` should be
+/// used to add additional LD contexts. A `build` context is added automatically. Returns after the
+/// LD client has been initialized or the `config_sync_timeout` timeout has been reached.
+pub async fn sync_ld_to_configset<F>(
```

Comment:
Having this be the only `pub` thing in the crate I think logically follows from how it works (it used to return a struct but...there was no point). But I don't love the API of "call this function, it returns ~nothing and probably syncs are being done". Thoughts? Is it fine? Other ideas?

Reply:
This is the API I'd have expected before seeing the PR, so seems fine to me.

If you really wanted, you could hand back the sync loop future (or `SyncedConfigSet` with a `pub async fn sync_loop`) and make the caller put it into a task.

Reply:
Agree, also the API I'd have expected.